packages/react-tabster/src/TabsterProvider.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { makeMergePropsCompat, useMergedRefs } from '@fluentui/react-utilities';
+import { getCurrentTabster, createTabster, Types as TabsterTypes } from 'tabster';
+import { TabsterContext, TabsterContextValue } from './TabsterContext';
+
+export interface TabsterProvideProps extends React.HTMLAttributes<HTMLElement> {
+  dir?: 'rtl' | 'ltr';
+  document?: Document;
+  /**
+   * The root is automatically set as the `body` element of the ownerDocument.
+   * This prop needs to be set if a custom root is used
+   */
+  customRoot?: boolean;
+}
+
+export interface TabsterProviderState extends TabsterProvideProps {
+  dir: 'ltr' | 'rtl';
+  document?: Document;
```

Comment:
```suggestion
  document: Document | undefined;
```

This should not be optional, in SSR it `document` will be `undefined`.

Reply:
Techinically speaking it's the same result 😅
But I agree that using `| undefined` is more explicit